src/main/java/com/sap/oss/phosphor/fosstars/data/github/GitHubCachingDataProvider.java

```diff
@@ -42,7 +42,7 @@ public boolean supports(Subject subject) {
 
   @Override
   protected final ValueSet fetchValuesFor(Subject subject) throws IOException {
-    return fetchValuesFor(cast(subject, GitHubProject.class));
+    return fetchValuesFor(castAndCopy(subject, GitHubProject.class));
```

Comment:
This update affects all data providers that extends `GitHubCachingDataProvider`. At lease it would result in multiple copies of subjects which increase memory usage. Although it may be safe for now, but do you have a strong reason to fix it in this class? As far as I understand, the problem is that `SecurityReview` has a subject. Would it be enough to make a shallow copy of a subject in `SecurityReviewsFromOpenSSF` and pass it to `SecurityReviews`s? It would break cycles, and would not affect other data providers.